platform/ios/PolyPodApp/PolyPod/HomeScreenView.swift

```diff
@@ -0,0 +1,653 @@
+    let imageName: String
+    let backgroundColor: Color
+    let buttonTitle: LocalizedStringKey
+    let buttonBackgroundColor: Color
+}
+
+protocol HomeScreenStorage {
+    var categoriesList: AnyPublisher<[HomeScreenSectionModel], Never> { get }
+}
+
+final class HomeScreenStorageAdapter: HomeScreenStorage {
+    let categoriesList: AnyPublisher<[HomeScreenSectionModel], Never>
+    let featureStorage: FeatureStorage
+    
+    init(featureStorage: FeatureStorage) {
+        self.featureStorage = featureStorage
+        self.categoriesList = featureStorage
+            .categoriesList
+            .map(HomeScreenStorageAdapter.mapCategoryModel)
+            .eraseToAnyPublisher()
+    }
+    
+    static func mapCategoryModel(_ categoryModels: [FeaturesCategoryModel]) -> [HomeScreenSectionModel] {
+        categoryModels.map { model in
+            HomeScreenSectionModel(title: model.name,
+                                   cards: mapToCards(model.features),
+                                   type: model.id)
+        }
+    }
+    
+    private static func mapToCards(_ features: [Feature]) -> [Card] {
+        features.map { feature in
+            Card(id: feature.id,
+                 title: feature.name,
+                 description: feature.description ?? "",
+                 image: UIImage(contentsOfFile: feature.thumbnail?.path ?? "") ?? UIImage(),
+                 backgroundColor: feature.thumbnailColor ?? .white)
+        }
+    }
+}
+
+final class HomeScreenViewModel: ObservableObject {
+    private var storage: HomeScreenStorage
+    private var storageCancellable: AnyCancellable?
+    @Published var sections: [HomeScreenSectionModel] = []
+    
+    init(storage: HomeScreenStorage) {
+        self.storage = storage
+    }
+    
+    func setup() {
+        storageCancellable = storage.categoriesList.sink { sections in
+            self.sections = sections
+        }
+    }
+}
+
+// MARK: - UI sizes
+
+struct HomeScreenConstants {
+    
+    struct Typography {
+        let font: Font
+        let alignment: TextAlignment
+    }
+    
+    struct Section {
+        static let verticalSpacing = PolyStyle.Spacing.plSpace8x
+        static let title = Typography(font: .custom(PolyStyle.Font.Family.jostMedium,
+                                                          size: PolyStyle.Font.Size.lg)
+                                        .weight(PolyStyle.Font.Weight.medium),
+                                      alignment: PolyStyle.Font.Alignment.left)
+    }
+    
+    struct View {
+        static let horizontalPadding = PolyStyle.Spacing.plSpace4x
+    }
+    
+    struct TileContainer {
+        static let verticalSpacing = PolyStyle.Spacing.plSpace3x
+        static let horizontalSpacing = PolyStyle.Spacing.plSpace3x
+        static let numberOfColumns = 3
+    }
+    
+    struct Tile {
+        static let cornerRadius = PolyStyle.Spacing.plSpace2x
+    }
+    
+    struct SmallTile {
```

Comment:
This seems like a whole lot of boilerplate, but I don't see an easy way to define structs dynamically in Swift. Guess they could be dictionaries, but I don't know what that could entail. Also, they don't have the same keys. Additionally, they simply don't follow the same protocol (or interface), so I really don't know. They at least seem to have a title as common feature.

Reply:
As mentioned in a another comment, the whole approach of styling will be revisited after we will release HomeScreen functionality, with the intent to be able to easily modify/share configuration; probably something like poly look;

Also with the experience of designing the HomeScreen on both iOS and Android, we did find that it would make a lot more sense to have it developed using React Native instead of doing it natively.